listings/applications/components/src/ownable.cairo

```diff
@@ -93,3 +96,146 @@ pub mod ownable_component {
+        set_contract_address(owner);
+        let (ownable, _) = deploy();
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    fn test_transfer_ownership_by_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+        let owner = contract_address_const::<'owner'>();
+        ownable.transfer_ownership(owner);
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_account_zero() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        ownable.transfer_ownership(Zero::zero());
+    }
+
+    #[test]
+    fn test_renounce_ownership() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        assert(ownable.owner() == Zero::zero(), 'not_zero_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_previous_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.renounce_ownership();
+    }
+
+    #[test]
+    fn test_ownership_renounced_event(){
+        let contract_address = contract_address_const::<'owner'>();
+        set_contract_address(contract_address);
+        let (ownable, address) = deploy();
+        ownable.renounce_ownership();
+        assert_eq!(
+            starknet::testing::pop_log(address),
+            Option::Some(Event::OwnershipRenouncedEvent(OwnershipRenouncedEvent { previous:contract_address }))
+        );
+    }
+
+    #[test]
+    fn test_ownership_transferred_event(){
+        let contract_address = contract_address_const::<'owner'>();
+        set_contract_address(contract_address);
+        let (ownable, address) = deploy();
+        let new_owner = contract_address_const::<'new_owner'>();
+        ownable.transfer_ownership(new_owner);
+        assert_eq!(
+            starknet::testing::pop_log(address),
+            Option::Some(Event::OwnershipTransferredEvent(OwnershipTransferredEvent { previous:contract_address, new:new_owner}))
```

Comment:
```suggestion
            Option::Some(MockContract::Event::OwnableEvent(OwnershipTransferredEvent { previous:contract_address, new:new_owner}.into()))
```

Reply:
This test can also be merged with `test_transfer_ownership_by_owner`, since they are testing the same flow.